data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/rule/RuleEvaluator.java

```diff
@@ -34,52 +34,39 @@ public class RuleEvaluator {
     private final TransformersFactory transformersFactory;
-    private String PLUGIN_NAME = null;
 
     public RuleEvaluator(TransformersFactory transformersFactory) {
         this.transformersFactory = transformersFactory;
     }
 
     public RuleEvaluatorResult isTransformationNeeded(PipelinesDataFlowModel pipelineModel) {
-        //TODO - Dynamically scan the rules folder and get the corresponding template.
-        return isDocDBSource(pipelineModel);
-    }
 
-    /**
-     * Evaluates model based on pre defined rules and
-     * result contains the name of the pipeline that will need transformation,
-     * evaluated boolean result and the corresponding template model
-     * Assumption: only one pipeline can have transformation.
-     *
-     * @param pipelinesModel
-     * @return RuleEvaluatorResult
-     */
-    private RuleEvaluatorResult isDocDBSource(PipelinesDataFlowModel pipelinesModel) {
-        PLUGIN_NAME = "documentdb";
-
-        Map<String, PipelineModel> pipelines = pipelinesModel.getPipelines();
+        Map<String, PipelineModel> pipelines = pipelineModel.getPipelines();
         for (Map.Entry<String, PipelineModel> entry : pipelines.entrySet()) {
             try {
                 String pipelineJson = OBJECT_MAPPER.writeValueAsString(entry);
-                if (evaluate(pipelineJson, PLUGIN_NAME)) {
-                    LOG.info("Rule for {} is evaluated true for pipelineJson {}", PLUGIN_NAME, pipelineJson);
+                RuleFileEvaluation ruleFileEvaluation = evaluate(pipelineJson);
+
+                if (ruleFileEvaluation.getResult()) {
+                    String pluginName = ruleFileEvaluation.getPluginName();
+                    LOG.info("Applying rule {}",ruleFileEvaluation.getRuleFileName().toString());
+                    LOG.info("Rule for {} is evaluated true for pipelineJson {}", pluginName, pipelineJson);
 
-                    InputStream templateStream = transformersFactory.getPluginTemplateFileStream(PLUGIN_NAME);
+                    InputStream templateStream = transformersFactory.getPluginTemplateFileStream(pluginName);
                     PipelineTemplateModel templateModel = yamlMapper.readValue(templateStream,
                             PipelineTemplateModel.class);
-                    LOG.info("Template is chosen for {}", PLUGIN_NAME);
+                    LOG.info("Template is chosen for {}", pluginName);
 
                     return RuleEvaluatorResult.builder()
                             .withEvaluatedResult(true)
                             .withPipelineTemplateModel(templateModel)
                             .withPipelineName(entry.getKey())
                             .build();
                 }
-            } catch (FileNotFoundException e){
-                LOG.error("Template File Not Found for {}", PLUGIN_NAME);
+            } catch (FileNotFoundException e) {
+                LOG.error("Template File Not Found");
```

Comment:
Why did you remove the additional information for tracking the problem?

Reply:
will add it back.